### PR TITLE
Boot support

### PR DESCRIPTION
**Pull request by patrickpichler**
I hope the way i implemented it is OK.

I'm relatively new to Clojure and would be glad for each comment on how to improve the code.

### CODE REVIEW

**Comment by patrickpichler** [comment]
@snoe can i get any feedback on this pull request?

**Comment by patrickpichler** [comment]
Thanks for your feedback, I will try to encoorparate that :+1:

**Comment by patrickpichler** [comment]
@snoe I now removed the record and replaced it with a map with two keys.

File: src/clojure_lsp/handlers.clj [hunk 1]
```diff
@@ -136,29 +136,58 @@
...
+  (lookup-classpath #(shell/sh "boot" "show" "--fake-classpath" :dir root-path) root-path))
+
+(defn build-lein-project [root-path]
+  (let [file-hash (hash-project-file (io/file root-path "project.clj"))
+        classpath (lein-lookup-classpath root-path)]
+    {:project-hash file-hash :classpath classpath}))
+
+(defn build-boot-project [root-path]
+  (let [file-hash (hash-project-file (io/file root-path "build.boot"))
+        classpath (lein-lookup-classpath root-path)]
```
Comment (snoe) [inline src/clojure_lsp/handlers.clj]: Should be `boot-lookup-classpath`

File: src/clojure_lsp/handlers.clj [hunk 2]
```diff
@@ -136,29 +136,58 @@
...
-      (-> (shell/sh "lein" "classpath" :dir root-path)
+    (let [sep (re-pattern (System/getProperty "path.separator"))]
+      (-> (command)
           (:out)
           (string/trim-newline)
           (string/split sep)))
     (catch Exception e
-      (log/warn "Could not run lein in" project-root (.getMessage e)))))
+      (log/warn "Error while looking up classpath info in" root-path (.getMessage e))
+      '[])))
```
Comment (snoe) [inline src/clojure_lsp/handlers.clj]: Vector doesn't need quoting

File: src/clojure_lsp/handlers.clj (hunk 1 above)
Comment (patrickpichler) [reply src/clojure_lsp/handlers.clj]: Thanks, that was an obvious one. I think the issue was that my tests solely consisted of seeing if I get some autocompletion for a boot project (which isn't enought, as this is done by nrepl I think).

**Comment by patrickpichler** [comment]
@snoe I changed the requested code parts :+1:

**Pull request merged**